pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BasePartitionUpsertMetadataManager.java

```diff
@@ -395,6 +402,18 @@ protected void finishOperation() {
     }
   }
 
+  /**
+   * When TTL is enabled for upsert, this function is used to remove expired keys from the primary key indexes.
+   */
+  @Override
+  public void removeExpiredPrimaryKeys(Comparable watermark) {
```

Comment:
Why do we pass in the watermark value instead of calculating it within the metadata manager? Who is going to manage the watermark?

Reply:
@Jackie-Jiang good point. if i keep the interface  `removeExpiredPrimaryKeys(Comparable watermark)` i need to call it within the metadata manager to access largestSeenComparisonValue - ttl.

Or i can call it in LLRealtimeDataManager with interface 
`removeExpiredPrimaryKeys()` since LLRealtimeDataManager do not have access of `largestSeenComparisonValue`.